Marlin/Marlin_main.cpp

```diff
@@ -5462,9 +5462,13 @@ inline void gcode_M303() {
     }
     switch(tool) {
         case 0:
-          OUT_WRITE(SYRINGE0_PIN, HIGH);
-          current_syringe_pin = SYRINGE0_PIN;
-          break;
+          if (!Cartridge__GetAugerEnabled()) {
+            OUT_WRITE(SYRINGE0_PIN, HIGH);
+            current_syringe_pin = SYRINGE0_PIN;
+            break;
+          }
+          SERIAL_PROTOCOLPGM("Auger enabled, syringe extension canceled");
+          return;
```

Comment:
Break should be outside the if, at the end of the case. I'd put this warning message in the `if` statement above and put the expected behavior here (i.e. switch them and remove the `!`). Same for `M383` below.